### PR TITLE
[IMP] project: Use stored computed fields for parent to children task

### PR DESCRIPTION
**Pull request by LucasLefevre**
Purpose
=======
When you create a task independently and link it to a Parent task later on,
several changes are applied: the person assigned on the sub-task switches
to the one set on the parent task, same for the customer, and so on.
This creates frustration because the user most likely set specific data on the
sub task for a reason. In addition, the changes are made implicitely, so the
user might not notice it or get confused as to what changed from the original
task.

Specification
=============
Values for fields `partner_id`, `email_from`, `sale_line_id` should be transfered
from parent task to children task, only if the value is not already set on the child.

This was already partially implemented by 749810a and 45396f7.
But to achieve it, similar code was duplicated in several methods:
default_get, onchanges, write, create.

Since the new ORM, the same behavior can be achieved with only computed fields
with store=True and readonly=False. This commit changes the previous implementation
to take advantage of this which greatly improves code readability and maintainability.

Tests by Maximilen Larue
Business code by Lucas Lefèvre

Co-authored-by: MaxLarue <mla@odoo.com>

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/project/models/project.py [hunk 1]
```diff
@@ -750,19 +718,20 @@ def update_date_end(self, stage_id):
...
-        parent_task = self.env['project.task'].browse(parent_id)
-        for field_name in self._subtask_default_fields():
-            result[field_name] = parent_task[field_name]
-        # special case for the subtask default project
-        result['project_id'] = parent_task.project_id.subtask_project_id
-        return self._convert_to_write(result)
+    @api.depends('parent_id.project_id.subtask_project_id')
+    def _compute_project_id(self):
+        for task in self:
+            task.project_id = task.project_id or task.parent_id.project_id.subtask_project_id
```
Comment (LucasLefevre) [inline addons/project/models/project.py]: ping @Julien00859 
```python
task.project_id = task.project_id or ...
```
?

Comment (Julien00859) [reply addons/project/models/project.py]: en général on ne réécris pas sur le field (la réécriture est effective), je préfère `if not task.project_id: task.project_id = task.parent_id.project_id.subtask_project_id`

**Comment by Julien00859** [comment]
+37 -102 lignes de code :rocket:
+216 lignes de test :rocket:

File: addons/project/models/project.py (hunk 1 above)
Comment (LucasLefevre) [reply addons/project/models/project.py]: ok merci :)

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**